lib/msf/core/post/linux/system.rb

```diff
@@ -227,27 +227,6 @@ def has_gcc?
     raise 'Unable to check for gcc'
   end
 
-  #
-  # Gets the process id(s) of `program`
-  # @return [Array]
-  #
-  def pidof(program)
-    pids = []
-    full = cmd_exec('ps aux').to_s
-    full.split("\n").each do |pid|
-      pids << pid.split(' ')[1].to_i if pid.include? program
-    end
-    pids
-  end
-
-  #
-  # Gets the uid of a pid
-  # @return [String]
-  #
-  def pid_uid(pid)
-    read_file("/proc/#{pid}/status").to_s
-  end
```

Comment:
This is referenced in `lib/msf/core/post/unix.rb` on L 102. That appears to be the only reference to it, but removing this function would break it there. This function doesn't do what the documentation says it does though. The contents of `/proc/#/status` is a text blob containing among other things, the UID but that still needs to be parsed out.

Because that's the only reference to it and it doesn't do what the function name or documentation implies, I'd be in favor of inlining this logic into `Msf::Post::Unix.whoami` and removing it.

Reply:
okay :+1: